OTP-P-Sample/OTPServer.p

```diff
@@ -0,0 +1,100 @@
+event Success;
+event Failure;
+
+event OTPSecretMsg: (machine, int);
+event OTPSecretReceived;
+event OTPCodeMsg: int;
+event OTPCodeValidated;
+event OTPCodeFailed;
+
+machine BANK_SERVER 
+{
+    var clientOtpGenerator: machine;
+	var secret: int;
+
+    // This is the entry point.
+    start state Init {
+        entry (payload:any) {
+          clientOtpGenerator = payload as machine;
+          raise (Success);   	   
+        }
+        on Success goto GenerateOTPSecret;
+    }
+
+    state GenerateOTPSecret {
+        entry {
+			// generate OTP secret 
+			secret = 123456788;
+			send clientOtpGenerator, OTPSecretMsg, (this, secret);
+	    }
+        on OTPSecretReceived goto WaitOTPCode;
+     }
+
+     state WaitOTPCode {
+        on OTPCodeMsg goto ValidateOTPCode;
+     }
+
+	 state ValidateOTPCode {
+        entry (payload: int) {
+		  // validate OTP code
+		  if (payload == 123456789) {
+			send clientOtpGenerator, OTPCodeValidated;
+          	raise (Success);  
+		  } else  {
+			send clientOtpGenerator, OTPCodeFailed;
+			raise (Failure);
+		  }  
+        }
+		on Success goto Done;
+		on Failure goto WaitOTPCode;
+     }
+
+     state Done {}
+}
+
+machine CLIENT_OTP_GENERATOR
+{
+	var bankServer: machine;
+	var OTPSecret: int;
+
+    start state Init {
+        on OTPSecretMsg goto HandleOTPSecret;
+    }
+
+    state HandleOTPSecret {
+	    entry (payload: (machine, int)) {
+	        bankServer = payload.0;
+			OTPSecret = payload.1;
+			send bankServer, OTPSecretReceived;
+			raise (Success);	 	  
```

Comment:
You can delete the success event and replace `raise(Success)` by `goto GenerateOTPCode;`